### DISCUSSION TITLE
GPU acceleration?

### DISCUSSION THREAD

**Discussion by Padmapalita** [vanderschaarlab/autoprognosis#76] [Q&A]
Hello, 
I've looked through the code for autoprognosis and it appears that you do use CUDA, but running on google cloud or my local GPU I can not see how to enable this? The documentation does not really mention it.

I've got a dataset of 57,000 rows and about 350 variables for a study I'm trying to run much in the same vein as the UK biobank publication for a master's project, so in terms of usability on the CPU auto prognosis is taking a few hours to do 1 itteration of a study. 

Any help or guidance would be hugely appreciated. 
thanks, 
Padma

**Comment by cheboluanusha** [comment]
> Hello, I've looked through the code for autoprognosis and it appears that you do use CUDA, but running on google cloud or my local GPU I can not see how to enable this? The documentation does not really mention it.
> 
> I've got a dataset of 57,000 rows and about 350 variables for a study I'm trying to run much in the same vein as the UK biobank publication for a master's project, so in terms of usability on the CPU auto prognosis is taking a few hours to do 1 itteration of a study.
> 
> Any help or guidance would be hugely appreciated. thanks, Padma

Is the issue regarding enabling CUDA for AutoPrognosis on Google Cloud or local GPU setups resolved? I'm encountering a similar challenge with my dataset, which contains 12,000 rows and 1,002 columns. Any advice or insights would be greatly appreciated.

**Reply by jonjinkim** [reply]
Thanks. I have the same issue which is that my local GPU is not being utilised.

**Reply by jonjinkim** [reply]
Thanks. This issue is resolved on my computer. Autoprognosis only uses GPUs for ML analysis, i.e. DeepHit etc. It does not use the GPUs for the other parts of the analysis or methods, eg Cox_ph.
